README.md

```diff
@@ -18,22 +18,109 @@ Pull Requests should meet a specific set of criteria before being merged.
+
+ - Add that PR to a queue dedicated to the branch this PR is targetting
+ - Post a comment in the PR to let you know that the PR has been queued and its position in the queue
+
+(†) Wall-E handles one merge queue per target branch to still allow PRs targetting different branches to be merged in parallel.
+
+
+### Integration of a PR
+
+When the bot dequeues the top PR from a queue, it will start its integration, which consists of the following steps:
+
+ - Merge the target branch back into the PR to ensure it's up-to-date if not
+ - Wait for its status checks to pass
+    - you can configure if you want the bot to only merge if _all_ statuses are green or only the _required_ ones via the `REQUIRES_ALL_STATUS_CHECKS` environment variable
+ - Once the PR is deemed ready (status checks pass, green on GitHub with the minimal number of approvals reached), it will merge the PR into the target branch (squash)
+ - Then it will go to the next PR in the queue
+
+If there is a failure at some point doring the integration of the PR – e.g. one of the (required) status check fails:
+
+ - the bot will post a comment with the error on the PR
+ - then it will remove the merge label
+ - and go to the next PR in the queue
+
+If you remove the merge label from a PR that was in the queue, that PR gets removed from the queue.
+
+### Top Priority PRs
+
+The bot also supports "Top Priority Labels" (configurable via the `TOP_PRIORITY_LABELS` environment variable)
+
+When you add one of such "Top Priority" labels to your PR, the bot will ensure that PRs marked as Top Priority will be merged before other PRs in the same queue, by making those PR jump at the front of the other non-top-priority PRs in the queue.
+
+For example, if your queue already contains PRs `A`,`B`,`C`,`D`,`E` with `A` and `B` already marked with one of the Top Priority label, then adding a Top Priority label to the PR `E` will make it jump in front of `C` and `D` but still after `A` and `B`, so the queue will become `A`,`B`,`E`,`C`,`D`.
+
+## Implementation details
+
+The whole codebase is implemented in Swift using Vapor.
+
+In you need to maintain/improve the code, here are some high-level implementation details that might help you navigate the codebase
+
+### MergeService
+
+`MergeService` is a service class representing a single merge queue.
+
+ - It handles the logic of the state machine for the various states and transitions to process each Pull Request in its queue in order
+ - It is implemented using [ReactiveFeedback](https://github.com/babylonhealth/ReactiveFeedback)
+
+Below are some diagrams to help you visualise the ReactiveFeedback state machine logic implemented in `MergeService`:
+
+<details>
+<summary>🔃 State Diagram</summary>
+
+![State Diagram](assets/mergebot_states.png)
+
+</details>
+
+<details>
+<summary>▶️ Action/Feedbacks Diagrams</summary>
+
+![Action/Feedbacks Diagrams](assets/mergebot_feedbacks.png)
```

Comment:
[`mergebot_feedbacks.png` ⤴️](https://raw.githubusercontent.com/babylonhealth/Wall-E/olivier/IOSP-515-CNSMR-3231/assets/mergebot_feedbacks.png)

Reply:
did you use one of these JS tools?